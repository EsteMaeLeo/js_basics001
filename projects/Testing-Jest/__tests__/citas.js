import Citas from "../js/classes/Citas";

describe("Test Citas Class", () => {
  const cita = new Citas();

  const id = Date.now();

  test("Add new Cita", () => {
    const citaObj = {
      mascota: "Doggy",
      propietario: "John",
      telefono: "65588771122",
      fecha: "10-10-2024",
      hora: "11:20",
      sintomas: "sick",
    };
    citaObj.id = id;

    cita.agregarCita(citaObj);

    //test
    expect(cita).toMatchSnapshot();
  });

  test("Update new Cita", () => {
    const citaUpdate = {
      id,
      mascota: "Charlie",
      propietario: "John",
      telefono: "65588771122",
      fecha: "10-10-2024",
      hora: "11:20",
      sintomas: "sick",
    };

    cita.editarCita(citaUpdate);

    //test
    expect(cita).toMatchSnapshot();
  });
});
