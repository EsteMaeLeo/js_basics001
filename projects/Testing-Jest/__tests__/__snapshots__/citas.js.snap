// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Citas Class Add new Cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-10-2024",
      "hora": "11:20",
      "id": 1704861142679,
      "mascota": "Doggy",
      "propietario": "John",
      "sintomas": "sick",
      "telefono": "65588771122",
    },
  ],
}
`;

exports[`Test Citas Class Delete new Cita 1`] = `
Citas {
  "citas": [],
}
`;

exports[`Test Citas Class Update new Cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-10-2024",
      "hora": "11:20",
      "id": 1704861142679,
      "mascota": "Charlie",
      "propietario": "John",
      "sintomas": "sick",
      "telefono": "65588771122",
    },
  ],
}
`;
